### PR TITLE
Added initial_url to the router component

### PR DESCRIPTION
**Pull request by norman784**
It seems that there was no way, nor workaround to set an active url when doing ssr, so I added a new prop to the router component.

### CODE REVIEW

File: packages/router/src/components/router.rs
Comment (rMazeiks) [inline packages/router/src/components/router.rs:32]: A bit more detail in the docs would be nice here...

Is this the initial URL? If so, `initial_url` might be a better name. As well as a note on what the default value would be.

Or is it a fixed URL that, when specified, cannot be changed with the push_route functions?

**Review by rMazeiks — APPROVED** [review]
Looks mostly good to me! Though it would be good if someone with more knowledge of the Router could take a look at this.

Thanks!

File: packages/router/src/components/router.rs
Comment (norman784) [reply packages/router/src/components/router.rs:32]: I'll update the docs and yes, `initial_url` is a better name.

This is intended to specify the initial url when working specifically with `ssr`, but I'm sure it will be helpful also for mobile and desktop, but I didn't tested those scenarios.

**Review by rMazeiks — APPROVED** [review]

**Comment by norman784** [comment]
@mrxiaozhuox are you available to review my PR?

**Comment by rMazeiks** [comment]
Note: there's a new, very big PR #522, a complete re-write of the router. I really like that PR, but unfortunately, merging it would make this one obsolete. Sorry!

(Let's see what the maintainers think of it – it's possible they won't want to make such breaking changes, and the new router would have to become a separate package – then, this PR is unaffected)

Not sure if the new router has something equivalent toe `initial_url`. If not, this code will probably be hard to integrate in it, and most likely will need to be re-written.

**Comment by norman784** [comment]
I though his work was stale, he just submitted yesterday his PR, I already checked it a while ago and his way on setting the initial url was more involved, but let's wait till his PR gets merge then.

**Review by jkelleyrtp — APPROVED** [review]

**Comment by jkelleyrtp** [comment]
Thanks!

I'm going to work on helping #522 get in but that'll take a bit longer than getting this one in now.

**Pull request merged**
